inventory_management_system_api/repositories/system.py

```diff
@@ -200,16 +195,16 @@ def _is_duplicate_system(
         )
         return system is not None
 
-    def _has_child_elements(self, system_id: CustomObjectId, session: Optional[ClientSession] = None) -> bool:
+    def has_child_elements(self, system_id: str, session: Optional[ClientSession] = None) -> bool:
```

Comment:
Is it worth doing the same to catalogue categories considering its the only one still using CustomObjectId?